packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -31,6 +36,103 @@ type BaseButtonWithIconProps = BaseButtonCommonProps & {
 
 export type BaseButtonProps = BaseButtonWithIconProps | BaseButtonWithoutIconProps;
 
+const StyledBaseText = styled(BaseText)(
+  ({
+    hasIcon,
+    iconPosition,
+    iconSpacing,
+  }: BaseTextProps &
+    Pick<BaseButtonProps, 'iconPosition'> & { hasIcon: boolean; iconSpacing: string }) => `
+    padding-left: ${hasIcon && iconPosition === 'left' ? iconSpacing : 0};
+    padding-right: ${hasIcon && iconPosition === 'right' ? iconSpacing : 0};
+`,
+);
+
+type ConfigProps = {
+  iconSize: IconSize; // prop
+  fontSize: keyof Theme['typography']['fonts']['size']; // prop
+  lineHeight: keyof Theme['typography']['lineHeights']; // prop
+  buttonHeight: '48px' | '40px' | '32px' | '28px'; // raw value
+  iconSpacing: string; // raw value
+  iconColor: IconProps['color']; // prop
+  textColor: BaseTextProps['color']; // prop
+  buttonSpacing: string; // raw value
+  text?: string;
+};
+
+const getProps = ({
+  theme,
+  size,
+  children,
+  buttonTypographyTokens,
+}: {
+  theme: Theme;
+  size: BaseButtonCommonProps['size'];
+  children?: string;
+  buttonTypographyTokens: typeof typography.onDesktop | typeof typography.onMobile;
+}): ConfigProps => {
+  const props: ConfigProps = {
+    iconSize: 'medium',
+    fontSize: buttonTypographyTokens.fonts.size.medium,
+    lineHeight: buttonTypographyTokens.lineHeight.medium,
+    buttonHeight: '40px',
+    iconSpacing: makeSpacingSize(theme.spacing[2]),
+    iconColor: 'action.icon.primary.default',
+    textColor: 'action.text.primary.default',
+    buttonSpacing: `${makeSpacingSize(theme.spacing[0])} ${makeSpacingSize(
+      theme.spacing[5],
+    )} ${makeSpacingSize(theme.spacing[0])} ${makeSpacingSize(theme.spacing[5])}`,
+    text: children?.trim(),
+  };
+
+  switch (size) {
+    case 'large':
+      props.iconSize = 'medium';
+      props.fontSize = buttonTypographyTokens.fonts.size.large;
+      props.lineHeight = buttonTypographyTokens.lineHeight.large;
+      props.buttonHeight = '48px';
+      props.iconSpacing = makeSpacingSize(theme.spacing[2]);
+      props.buttonSpacing = `${makeSpacingSize(theme.spacing[0])} ${makeSpacingSize(
+        theme.spacing[5],
+      )} ${makeSpacingSize(theme.spacing[0])} ${makeSpacingSize(theme.spacing[5])}`;
+      break;
```

Comment:
nit: add new line

Reply:
After every `break`? Won't that look odd?